### PR TITLE
fix: Don't require full tab before payment

### PR DESCRIPTION
**Pull request by jmcodingde**
## Summary

- With pending purchases, tabs don't actually become full, anymore.
- This PR removes the check for a full tab in `#payTab()`.

## Context

The introduction of the pending purchase feature in tapper broke the checkout flow in tab.js because it is using `supertab-browser`.

## Changes

- fix: Don't require tab to be full before payment
- fix: The tabs endpoint doesn't return full tabs, anymore

## Refs

- Tapper "Pending Purchase" PR: https://github.com/laterpay/tapper-core/pull/4887
- Related Slack conversation: https://supertab.slack.com/archives/C044ZLL80E9/p1727436031903729?thread_ts=1727349805.688419&cid=C044ZLL80E9

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -278,9 +278,7 @@ export class Supertab {
       paymentModel: "pay_later",
     });
 
-    const filterStatuses: TabStatus[] = [TabStatus.Open, TabStatus.Full];
-
-    if (filterStatuses.includes(tab?.status)) {
+    if (tab?.status === TabStatus.Open) {
```
Comment (andreagrandi) [inline src/index.ts:281]: @jmcodingde I know absolutely nothing about TS, but I feel this change will only include `OPEN` tab, is it correct? What about "legacy" tabs (those which have been closed but never paid) ?

Comment (jmcodingde) [reply src/index.ts]: Reverted that change here: [4bb919f](https://github.com/laterpay/supertab-browser/pull/75/commits/4bb919f4eb11339d73f0a1c5e9e49e5ba6d56a3a)

**Review by andreagrandi — APPROVED** [review]
🚀

**Pull request merged**